Sources/Operator/local_liouvillian.hpp

```diff
@@ -0,0 +1,143 @@
+//
+// Created by Filippo Vicentini on 08/11/2019.
+//
+
+#ifndef NETKET_LOCAL_LIOUVILLIAN_HPP
+#define NETKET_LOCAL_LIOUVILLIAN_HPP
+
+#include <utility>
+
+#include "Graph/doubled_graph.hpp"
+#include "Hilbert/abstract_hilbert.hpp"
+#include "Hilbert/doubled_hilbert.hpp"
+#include "Operator/abstract_operator.hpp"
+#include "Operator/local_operator.hpp"
+
+namespace netket {
+/**
+ * This struct represents a non-zero matrix-element L(v,v_row', v_col') of a
+ * super-operator for a given visible state v=(v_row, v_col).
+ */
+struct ConnectorSuperopRef {
+  /// The matrix element L(v_row, v_col ,v_row', v_col')
+  Complex mel;
+  /// The indices at which v_row needs to be changed to obtain v_row'
+  nonstd::span<const int> tochange_row;
+  /// The new values such that
+  ///    v'(tochange_row[k]) = newconf_row[k]
+  /// and v'(i) = v(i) for i ∉ tochange_row.
+  nonstd::span<const double> newconf_row;
+
+  /// The indices at which v_col needs to be changed to obtain v_col'
+  nonstd::span<const int> tochange_col;
+  nonstd::span<const double> newconf_col;
+};
+
+/**
+      Class describing the Liouvillian of a Lindblad Master Equation for
+      Open Quantum systems
+*/
+class LocalLiouvillian : public AbstractOperator {
+ public:
+  using MelType = std::vector<Complex>;
+  using MatType = std::vector<std::vector<MelType>>;
+  using SiteType = std::vector<int>;
+  using MapType = std::map<std::vector<double>, int>;
+  using StateType = std::vector<std::vector<double>>;
+  using ConnType = std::vector<std::vector<int>>;
+  using VectorType = AbstractOperator::VectorType;
+  using VectorRefType = AbstractOperator::VectorRefType;
+  using VectorConstRefType = AbstractOperator::VectorConstRefType;
+  using ConnSuperOpCallback = std::function<void(ConnectorSuperopRef)>;
+
+ private:
+  // The effective non hermitian hamiltonian Hnh_ = H_ - im/2\sum_i L_i^dag L_i
+  LocalOperator Hnh_;
+  LocalOperator H_;
+  std::vector<LocalOperator> jump_ops_;
+
+  // Adjoint of Hnh_
+  LocalOperator Hnh_dag_;
+
+ public:
+  /**
+   * Constructs a LocalLiouvillian from the hamiltonian H and with no jump
+   * operators. The Hamiltonian must be a LocalOperator (other types of
+   * operators are not supported).
+   * @param H : the Hamiltonian
+   */
+  explicit LocalLiouvillian(const LocalOperator &H);
+
+  /**
+   * Constructs a LocalLiouvillian from the hamiltonian H and the vector of
+   * jump operators ops. The Hamiltonian must be a LocalOperator (other types
+   * of operators are not supported).
+   * @param H : the Hamiltonian.
+   * @param jump_ops : a vector of LocalOperators.
+   */
+  explicit LocalLiouvillian(const LocalOperator &H,
+      const std::vector<const LocalOperator> &jump_ops);
```

Comment:
@femtobit Could you point me in the right direction in how to convert a python list (containing only LocalOperators) to a vector of const in order to bind this constructor to python?

Reply:
NVM, It's just a compilation problem. I figured it out..